src/process-cache.js

```diff
@@ -53,9 +31,7 @@ export default async function processCache(
         }
     }
 
-    if (json) {
-        return outputJson(log, jsonItems);
-    }
+    const output = getOutput(options);
 
-    return outputText(options, violationFileNames, log);
+    return output(options, log, jsonItems, violationFileNames);
```

Comment:
I really like the elegance of this code now. It feels "right" on a fundamental level, even if we know there are more architecture changes we want. Thanks for taking the time to do this.